dynamic-datasource-spring-boot3-starter/src/test/java/com/baomidou/dynamic/datasource/fixture/v3/AddRemoveDatasourceTest.java

```diff
@@ -53,6 +53,13 @@ void testAddAndRemoveDataSource() {
         dataSourceProperty.setUrl("jdbc:h2:mem:test1");
         dataSourceProperty.setDriverClassName("org.h2.Driver");
         DynamicRoutingDataSource ds = (DynamicRoutingDataSource) dataSource;
+        // async destroy datasource
+        ds.setGraceDestroy(true);
+        ds.addDataSource(dataSourceProperty.getPoolName(), dataSourceCreator.createDataSource(dataSourceProperty));
+        assertThat(ds.getDataSources().keySet()).contains("slave_1");
+        ds.removeDataSource("slave_1");
+        // close directly
+        ds.setGraceDestroy(false);
```

Comment:
- Just a reminder, you can create a new unit test class or in a new unit test function body to narrow the scope of your unit tests.

- It seems that the two codes in the unit test you changed are missing assertions.